src/components/FindMovie/FindMovie.tsx

```diff
@@ -20,38 +73,84 @@ export const FindMovie: React.FC = () => {
-            Can&apos;t find a movie with such a title
-          </p>
+          {error && (
+            <p className="help is-danger" data-cy="errorMessage">
+              Can&apos;t find a movie with such a title
+            </p>
+          )}
         </div>
 
         <div className="field is-grouped">
           <div className="control">
-            <button
-              data-cy="searchButton"
-              type="submit"
-              className="button is-light"
-            >
-              Find a movie
-            </button>
-          </div>
+            {movie ? (
+              <button
+                data-cy="searchButton"
+                type="submit"
+                className={classNames(
+                  'button',
+                  'is-ligth',
+                  {
+                    'is-loading': spinner,
+                  },
+                )}
+                disabled={query.length === 0}
+                onClick={() => setSpinner(true)}
+              >
+                Search again
+              </button>
+            ) : (
+              <button
+                data-cy="searchButton"
+                type="submit"
+                className={classNames(
+                  'button',
+                  'is-ligth',
+                  {
+                    'is-loading': spinner,
+                  },
+                )}
+                disabled={query.length === 0}
+                onClick={() => setSpinner(true)}
+              >
+                Find a movie
+              </button>
+            )}
 
-          <div className="control">
-            <button
-              data-cy="addButton"
-              type="button"
-              className="button is-primary"
-            >
-              Add to the list
-            </button>
           </div>
+          {movie !== null && (
+            <div className="control">
+              <button
+                data-cy="addButton"
+                type="button"
+                className="button is-primary"
+                onClick={() => {
+                  setMovies(current => {
+                    if (current
+                      .find(curMovie => curMovie.imdbId === movie.imdbId)) {
+                      setMovie(null);
+
+                      return [...current];
+                    }
+
+                    return [...current, movie];
+                  });
+                  setQuery('');
+                  setMovie(null);
+                }}
+              >
+                Add to the list
+              </button>
+            </div>
+          )}
         </div>
       </form>
 
-      <div className="container" data-cy="previewContainer">
-        <h2 className="title">Preview</h2>
-        {/* <MovieCard movie={movie} /> */}
-      </div>
+      {movie && (
+        <div className="container" data-cy="previewContainer">
+          <h2 className="title">Preview</h2>
+          {movie && <MovieCard movie={movie} />}
```

Comment:
you already do this check before
```suggestion
          <MovieCard movie={movie} />
```